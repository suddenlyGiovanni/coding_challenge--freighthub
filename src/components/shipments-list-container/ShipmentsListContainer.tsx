import * as React from 'react'
import styled from '@emotion/styled/macro'
import Pagination from 'material-ui-flat-pagination'

import { Shipments } from 'features/shipments/typings'
import { ShipmentsListElement } from 'components'

const ListContainer = styled.ul`
  position: relative;

  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: flex-start;

  width: 100%;
  margin-top: 30px;

  list-style-type: unset;

  margin-block-start: 0;
  margin-block-end: 0;
  margin-inline-start: 0;
  margin-inline-end: 0;
  padding-inline-start: 0;
`

const PaginationStyled = styled(Pagination)`
  text-align: center;
`

export const Page: React.FC<{
  shipments: Shipments
  onSelectedShipment: (shipmentId: string) => void
}> = ({ shipments, onSelectedShipment }) => {
  return (
    <ListContainer>
      {shipments &&
        shipments.map(shipment => (
          <ShipmentsListElement
            key={shipment.id}
            shipment={shipment}
            onClick={onSelectedShipment}
          />
        ))}
    </ListContainer>
  )
}

interface Props {
  shipments: Shipments[]
  onSelectedShipment: (shipmentId: string) => void
}

export const ShipmentsListContainer: React.FC<Props> = ({
  shipments,
  onSelectedShipment,
}) => {
  const [offset, setOffset] = React.useState(0)
  const pages = shipments.length

  return (
    <div>
      <PaginationStyled
        limit={1}
        offset={offset}
        total={pages}
        onClick={(e, offset) => setOffset(offset)}
      />

      {shipments && (
        <Page
          shipments={shipments[offset]}
          onSelectedShipment={onSelectedShipment}
        />
      )}
    </div>
  )
}
