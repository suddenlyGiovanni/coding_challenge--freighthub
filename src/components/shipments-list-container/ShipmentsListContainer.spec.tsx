import React from 'react'
import { render } from '@testing-library/react'

import { ShipmentsListContainer, Props } from './ShipmentsListContainer'
import { shipments } from 'test/mockDb'

describe('ShipmentsListContainer', () => {
  const props: Props = {
    shipments: [shipments],
    onSelectedShipment: shipmentId => {},
  }

  test('snapshot', () => {
    // Arrange
    const { container } = render(<ShipmentsListContainer {...props} />)
    // Assert
    expect(container).toMatchSnapshot()
  })

  test('renders no items when the shipment list is empty', () => {
    const { queryByTestId } = render(
      <ShipmentsListContainer {...props} shipments={[]} />
    )
    expect(queryByTestId('page-container')).toBeNull()
  })
})
