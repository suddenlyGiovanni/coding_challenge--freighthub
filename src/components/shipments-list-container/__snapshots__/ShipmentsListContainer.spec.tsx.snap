// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShipmentsListContainer snapshot 1`] = `
<div>
  <div>
    <div
      class="MuiFlatPagination-root css-1l5gfa2-PaginationStyled efufphl1"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiFlatPageButton-root MuiFlatPageButton-rootEnd MuiFlatPagination-rootEnd MuiButton-text MuiFlatPageButton-text MuiFlatPagination-text MuiButton-textPrimary MuiFlatPageButton-textPrimary MuiFlatPagination-textPrimary Mui-disabled Mui-disabled Mui-disabled Mui-disabled"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="MuiButton-label MuiFlatPageButton-label MuiFlatPagination-label"
        >
          &lt;
        </span>
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiFlatPageButton-root MuiFlatPageButton-rootCurrent MuiFlatPagination-rootCurrent MuiButton-text MuiFlatPageButton-text MuiFlatPagination-text MuiButton-textSecondary MuiFlatPageButton-textSecondary MuiFlatPagination-textSecondary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label MuiFlatPageButton-label MuiFlatPagination-label"
        >
          1
        </span>
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiFlatPageButton-root MuiFlatPageButton-rootEnd MuiFlatPagination-rootEnd MuiButton-text MuiFlatPageButton-text MuiFlatPagination-text MuiButton-textPrimary MuiFlatPageButton-textPrimary MuiFlatPagination-textPrimary Mui-disabled Mui-disabled Mui-disabled Mui-disabled"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="MuiButton-label MuiFlatPageButton-label MuiFlatPagination-label"
        >
          &gt;
        </span>
      </button>
    </div>
    <ul
      class="css-j27k6n-ListContainer efufphl0"
    >
      <li
        class="css-1l57fd2-ListElementContainer e8me0d10"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
        >
          <button
            class="MuiButtonBase-root MuiCardActionArea-root"
            data-testid="shipment-list-item"
            tabindex="0"
            type="button"
          >
            <div
              class="MuiCardContent-root css-1pomfrd-StyledCardContent e8me0d11"
            >
              <div
                class="css-f4l74i-HeadingContainer e8me0d12"
              >
                <div
                  class="css-fue0wc-HeadingDetail e8me0d13"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    data-testid="ship-icon"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M20 21c-1.39 0-2.78-.47-4-1.32-2.44 1.71-5.56 1.71-8 0C6.78 20.53 5.39 21 4 21H2v2h2c1.38 0 2.74-.35 4-.99 2.52 1.29 5.48 1.29 8 0 1.26.65 2.62.99 4 .99h2v-2h-2zM3.95 19H4c1.6 0 3.02-.88 4-2 .98 1.12 2.4 2 4 2s3.02-.88 4-2c.98 1.12 2.4 2 4 2h.05l1.89-6.68c.08-.26.06-.54-.06-.78s-.34-.42-.6-.5L20 10.62V6c0-1.1-.9-2-2-2h-3V1H9v3H6c-1.1 0-2 .9-2 2v4.62l-1.29.42c-.26.08-.48.26-.6.5s-.15.52-.06.78L3.95 19zM6 6h12v3.97L12 8 6 9.97V6z"
                    />
                  </svg>
                  <p
                    class="MuiTypography-root css-17328ce-Heading e8me0d14 MuiTypography-body1"
                  >
                    S1001
                  </p>
                </div>
                <div
                  class="css-fue0wc-HeadingDetail e8me0d13"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2c-4.97 0-9 4.03-9 9 0 4.17 2.84 7.67 6.69 8.69L12 22l2.31-2.31C18.16 18.67 21 15.17 21 11c0-4.97-4.03-9-9-9zm0 2c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.3c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                  <p
                    class="MuiTypography-root css-17328ce-Heading e8me0d14 MuiTypography-body1"
                  >
                    U1002
                  </p>
                </div>
                <div
                  class="css-fue0wc-HeadingDetail e8me0d13"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M5 4v3h5.5v12h3V7H19V4z"
                    />
                    <path
                      d="M0 0h24v24H0V0z"
                      fill="none"
                    />
                  </svg>
                  <p
                    class="MuiTypography-root css-17328ce-Heading e8me0d14 MuiTypography-body2 MuiTypography-noWrap"
                  >
                    New spring collection
                  </p>
                </div>
              </div>
              <div
                class="css-a14iro-TrackingContainer e8me0d15"
              >
                <div
                  class="css-i6u61p-OverviewContainer e8gzhee0"
                >
                  <div
                    class="css-1satmnw-LocationContainer e1cot6dd0"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-colorPrimary"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M15 11V5l-3-3-3 3v2H3v14h18V11h-6zm-8 8H5v-2h2v2zm0-4H5v-2h2v2zm0-4H5V9h2v2zm6 8h-2v-2h2v2zm0-4h-2v-2h2v2zm0-4h-2V9h2v2zm0-4h-2V5h2v2zm6 12h-2v-2h2v2zm0-4h-2v-2h2v2z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                    <p
                      class="MuiTypography-root css-192817u-PlaceName e1cot6dd1 MuiTypography-body2 MuiTypography-noWrap"
                      data-testid="place-name"
                      style="left: 0px;"
                    >
                      Ningbo port
                    </p>
                  </div>
                  <div
                    class="css-1lfx6p1-Temp e8gzhee1"
                  >
                    <div
                      aria-valuenow="50"
                      class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate"
                      role="progressbar"
                      style="margin: 0px 10px;"
                    >
                      <div
                        class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
                        style="transform: translateX(-50%);"
                      />
                    </div>
                  </div>
                  <div
                    class="css-1satmnw-LocationContainer e1cot6dd0"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M15 11V5l-3-3-3 3v2H3v14h18V11h-6zm-8 8H5v-2h2v2zm0-4H5v-2h2v2zm0-4H5V9h2v2zm6 8h-2v-2h2v2zm0-4h-2v-2h2v2zm0-4h-2V9h2v2zm0-4h-2V5h2v2zm6 12h-2v-2h2v2zm0-4h-2v-2h2v2z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                    <p
                      class="MuiTypography-root css-192817u-PlaceName e1cot6dd1 MuiTypography-body2 MuiTypography-noWrap"
                      data-testid="place-name"
                      style="right: 0px;"
                    >
                      Saarbr√ºcker Str. 38, 10405 Berlin
                    </p>
                  </div>
                </div>
              </div>
            </div>
            <span
              class="MuiCardActionArea-focusHighlight"
            />
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </li>
      <li
        class="css-1l57fd2-ListElementContainer e8me0d10"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
        >
          <button
            class="MuiButtonBase-root MuiCardActionArea-root"
            data-testid="shipment-list-item"
            tabindex="0"
            type="button"
          >
            <div
              class="MuiCardContent-root css-1pomfrd-StyledCardContent e8me0d11"
            >
              <div
                class="css-f4l74i-HeadingContainer e8me0d12"
              >
                <div
                  class="css-fue0wc-HeadingDetail e8me0d13"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    data-testid="plane-icon"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M21 16v-2l-8-5V3.5c0-.83-.67-1.5-1.5-1.5S10 2.67 10 3.5V9l-8 5v2l8-2.5V19l-2 1.5V22l3.5-1 3.5 1v-1.5L13 19v-5.5l8 2.5z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                  <p
                    class="MuiTypography-root css-17328ce-Heading e8me0d14 MuiTypography-body1"
                  >
                    S1002
                  </p>
                </div>
                <div
                  class="css-fue0wc-HeadingDetail e8me0d13"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2c-4.97 0-9 4.03-9 9 0 4.17 2.84 7.67 6.69 8.69L12 22l2.31-2.31C18.16 18.67 21 15.17 21 11c0-4.97-4.03-9-9-9zm0 2c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.3c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                  <p
                    class="MuiTypography-root css-17328ce-Heading e8me0d14 MuiTypography-body1"
                  >
                    U1001
                  </p>
                </div>
                <div
                  class="css-fue0wc-HeadingDetail e8me0d13"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M5 4v3h5.5v12h3V7H19V4z"
                    />
                    <path
                      d="M0 0h24v24H0V0z"
                      fill="none"
                    />
                  </svg>
                  <p
                    class="MuiTypography-root css-17328ce-Heading e8me0d14 MuiTypography-body2 MuiTypography-noWrap"
                  >
                    PO89634, PO27X
                  </p>
                </div>
              </div>
              <div
                class="css-a14iro-TrackingContainer e8me0d15"
              >
                <div
                  class="css-i6u61p-OverviewContainer e8gzhee0"
                >
                  <div
                    class="css-1satmnw-LocationContainer e1cot6dd0"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-colorPrimary"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M15 11V5l-3-3-3 3v2H3v14h18V11h-6zm-8 8H5v-2h2v2zm0-4H5v-2h2v2zm0-4H5V9h2v2zm6 8h-2v-2h2v2zm0-4h-2v-2h2v2zm0-4h-2V9h2v2zm0-4h-2V5h2v2zm6 12h-2v-2h2v2zm0-4h-2v-2h2v2z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                    <p
                      class="MuiTypography-root css-192817u-PlaceName e1cot6dd1 MuiTypography-body2 MuiTypography-noWrap"
                      data-testid="place-name"
                      style="left: 0px;"
                    >
                      Shanghai Port
                    </p>
                  </div>
                  <div
                    class="css-1lfx6p1-Temp e8gzhee1"
                  >
                    <div
                      aria-valuenow="100"
                      class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate"
                      role="progressbar"
                      style="margin: 0px 10px;"
                    >
                      <div
                        class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
                        style="transform: translateX(0%);"
                      />
                    </div>
                  </div>
                  <div
                    class="css-1satmnw-LocationContainer e1cot6dd0"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-colorPrimary"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M15 11V5l-3-3-3 3v2H3v14h18V11h-6zm-8 8H5v-2h2v2zm0-4H5v-2h2v2zm0-4H5V9h2v2zm6 8h-2v-2h2v2zm0-4h-2v-2h2v2zm0-4h-2V9h2v2zm0-4h-2V5h2v2zm6 12h-2v-2h2v2zm0-4h-2v-2h2v2z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                    <p
                      class="MuiTypography-root css-192817u-PlaceName e1cot6dd1 MuiTypography-body2 MuiTypography-noWrap"
                      data-testid="place-name"
                      style="right: 0px;"
                    >
                      Saarbr√ºcker Str. 38, 10405 Berlin
                    </p>
                  </div>
                </div>
              </div>
            </div>
            <span
              class="MuiCardActionArea-focusHighlight"
            />
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
